File: COURS/M1/SEMESTRE2/PROG/PROJET_ELIOT/Animal.java
public class Animal {
    private String nom;
    private int age;
    private boolean vivant;

    public Animal() {
        nom = "Absent";
        age = 0;
        vivant = true;
    }

    public Animal(String nom, int age) {
        this.nom = nom;
        this.age = age;
        vivant = true;
    }

    public Animal(String nom) {
        this.nom = nom;
        this.age = 0;
        vivant = true;
    }

    public void affiche() {
        System.out.println("Je m'apelle " + this.nom);
        System.out.println("J'ai " + this.age + " ans!");
    }

}

File: COURS/M1/SEMESTRE2/POO/TD1-2/Animal.cpp
//
// Created by eliot on 08/01/19.
//

#include "Animal.h"
//#include <string>
//using namespace std;

Animal::Animal(string nom, int age) : nom(move(nom)), age(age), vivant(true) {}

int Animal::getAge() {
    return age;
}

void Animal::setAge(int year) {
    this->age = year;
}

void Animal::die() {
    vivant = false;
}
